Creación de clase MateriaData

gregorionunez8 opened the pull request:
Clase MateriaData para:
- Agregar una materia
- Modificar una materia
- Buscar una materia por IdMateria

gregorionunez8 committed:
Update Inscripcion.class

gregorionunez8 committed:
Update MateriaData.java

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -0,0 +1,96 @@
+        }
+        return materia;    
+    }
+
```
Agregar los metodos eliminarMateria por id y listarMaterias

gregorionunez8 replied:
Listo!

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -0,0 +1,96 @@
+        String sql = "INSERT INTO materia(nombre, anio, estado) VALUES (?, ?, ?)";
+        
+        try {
+            Conexion.getConexion(); //Conexion con la base de datos
```
creo que este esta de mas

gregorionunez8 replied:
Elimine la conexión del constructor y la hice adentro del método.

gregorionunez8 committed:
Creacion de la clase MateriaData

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -44,18 +45,17 @@ public void agregarMateria(Materia materia) {
+
     // Metodo para modificar materia
-    public void modificarMateria(Materia materia){
+    public void modificarMateria(Materia materia, Materia materia2) {
```
El metodo debe recibir un solo parametro de tipo Materia

gregorionunez8 replied:
Listo!

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -66,31 +66,73 @@ public void modificarMateria(Materia materia){
+    // Metodo para eliminar materia
+    public void eliminarMateria(int id) {
+        // Consulta sql para insertar materia
+        String sql = "DELETE FROM materia WHERE idMateria=?";
```
Debe ser un UPDATE ya que se pide borrado logico

gregorionunez8 replied:
Listo!

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -66,31 +66,73 @@ public void modificarMateria(Materia materia){
+            PreparedStatement ps = con.prepareStatement(sql); //PreparedStatement con la consulta sql
+            ps.setInt(1, id); //Asignacion de valores
+            ps.executeUpdate(); // Ejecutar PreparedStatement
+            JOptionPane.showMessageDialog(null, "Materia eliminada");
```
El mensaje no debe aparecer cuando la materia no existe

gregorionunez8 replied:
Listo!

gregorionunez8 committed:
Correccion de MateriaData

-Metodo materiaExiste para corroborar la existencia de la materia
-Modificacion a eliminado logico
-Correccion de modificarMateria
